spec.md

```diff
@@ -547,6 +547,19 @@ Analog<32> ; 32-bit analog type
 Analog     ; analog type with inferred width
 ```
 
+### Enumeration Types
+
+Enumerations are structural disjoint union types.  An enumeration has a number
+of variants, each with a type.  The different variants are specified with tags.
+A variant may optionally omit the data type, in which case it is implicitly
+defined to be `UInt<0>`. The variant types of an enumeration must all be
+passive and cannot contain analog types.
+
+``` firrtl
+[#a, #b, #c]
+[#some: UInt<8>, #none]
```

Comment:
Final bikesheding round:
`{{ some : UInt<8>, none}}` or `{| some : UInt<8>, none |}` or `{# some : UInt<8>, none #}`
Why?  These are closest in structure to structs and I don't like this being the one place field identifiers are prefixed.
(hash is sometimes associated with comments, so I'd prefer not that form.  Those examples are I think in order of bikeshed preference)